src/scales/scale.time.js

```diff
@@ -139,7 +139,7 @@ function lookup(table, key, value) {
 
 	while (lo >= 0 && lo <= hi) {
 		mid = (lo + hi) >> 1;
-		i0 = table[mid - 1] || null;
+		i0 = mid > 0 && table[mid - 1] || null;
```

Comment:
It seems to me like this can only happen when `lo` and `hi` both equal 0. Is that right? Why do we need a special check for that? It would seem the code comment (`given value is outside table (before first item)`) no longer applies in that case

I think we should get rid of the interpolation stuff entirely, which would be much faster and simpler. We can instead check the `distribution` to be a lot more efficient. Something like:

```
getPixelForValue(value) {
	const me = this;
	if (distribution === 'series') {
		return getPixelForDecimal(me._timestamps.indexOf(value) / me._timestamps.length);
	} else {
		return getPixelForDecimal((value - me.max) / (me.min - me.max));
	}
}
```

Reply:
This happens when lo = 0 and hi = 1. `(0 + 1) >> 1` = 0
`indexOf` would be a lot slower than the lookup table.

Reply:
Another problem with indexOf is that you have to have all the timestamps in place to be able to get a pixel for it.
For example:
```js
data: [{x: 1577659670, y: 1}, {x: 1577659680, y: 2}]

getPixlForValue(1577659675) // undefined 
```

Reply:
Yeah, you're right that `indexOf` would be a `O(n)` lookup. We could use a `Map` instead to make it `O(1)`. Especially since you're already creating a `Map` in https://github.com/chartjs/Chart.js/pull/6871 (though we'd need the value to be the index rather than `true`). An easy first implementation would just be to remove the `only add points that breaks the scale linearity` check in `buildLookupTable` so that every value is present. Then we could just call `lookup` instead of `interpolate`, which I'm pretty sure would still be faster than what we do today. Though that would be `O(log n)` instead of the `O(1)` lookup we could get using a `Map`

I don't think that we'd ever call `getPixelForValue` for a value that's not in the dataset provided by the user, so I'd be fine with explicitly not supporting that as I don't see a use case for it.